diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1af95a8..9d1c562 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -47,16 +47,16 @@ endif ()
 include(utility)
 
 # set up option variable for cmake
-option(BUILD_RTTR_DYNAMIC "Build the dynamic/shared version of RTTR library" ON)
-option(BUILD_UNIT_TESTS "Build the unit tests of RTTR" ON)
-option(BUILD_STATIC "Build RTTR as static library" OFF)
+option(BUILD_RTTR_DYNAMIC "Build the dynamic/shared version of RTTR library" OFF)
+option(BUILD_UNIT_TESTS "Build the unit tests of RTTR" OFF)
+option(BUILD_STATIC "Build RTTR as static library" ON)
 option(BUILD_WITH_STATIC_RUNTIME_LIBS "Link against the static runtime libraries" OFF)
 option(BUILD_WITH_RTTI "Enable build with C++ runtime type information for compilation" ON)
 option(BUILD_BENCHMARKS "Enable this to build the benchmarks" OFF)
-option(BUILD_EXAMPLES "Enable this to build the examples" On)
-option(BUILD_DOCUMENTATION "Enable this to build the documentation" ON)
-option(BUILD_INSTALLER "Enable this to build the installer" ON)
-option(BUILD_PACKAGE "Enable this to build the installer" ON)
+option(BUILD_EXAMPLES "Enable this to build the examples" OFF)
+option(BUILD_DOCUMENTATION "Enable this to build the documentation" OFF)
+option(BUILD_INSTALLER "Enable this to build the installer" OFF)
+option(BUILD_PACKAGE "Enable this to build the installer" OFF)
 option(USE_PCH "Use precompiled header files for compilation" ON)
 option(CUSTOM_DOXYGEN_STYLE "Enable this option to use a custom doxygen style for HTML documentation; Otherwise the default will be used" ON)
 option(BUILD_WEBSITE_DOCU "Enable this option to create the special docu for the website" OFF)
